app/models/project.rb

```diff
@@ -13,13 +13,17 @@ class Project < ActiveRecord::Base
 
   after_create :create_template_defaults
 
-  attr_accessible :name, :description, :template, :archived
+  attr_accessible :name, :description, :template, :archived, :extensions
 
   scope :active,
     :include    => [:users, {:stages => [:hosts, :recipes]}, :configuration_parameters, :user_project_links],
     :conditions => {:archived => false},
     :order      => "projects.name ASC"
 
+  serialize :extensions
+
+  AVAILABLE_EXTENSIONS = %w{rvm bundler rails/assets rails/migrations rbenv composer symfony npm laravel chruby}
```

Comment:
This list has to stay co-ordinated with the Gemfile. That means you have to do bookkeeping, and a change in one place can lead to a break in the other. I hate bookkeeping, especially in Ruby. It would be nice if both the Gemfile (which is, at the end of the day, Ruby code) and this class would load the extensions from the same common file. Since the mappings aren't quite one-to-one, you probably want a hash of gem names onto values that are either a single element or a list of extensions provided by that gem. The gemfile will use the keys, and this file will use the values. (Do `Array[value].flatten` to ensure the values are all arrays.)